index.js

```diff
@@ -57,7 +60,12 @@ function forwardResponseToApiGateway(server, response, context) {
 
             Object.keys(headers)
                 .forEach(h => {
-                    if(Array.isArray(headers[h])) headers[h] = headers[h].join(',')
+                    if(Array.isArray(headers[h])) {
+                      headers[h].forEach((value, i) => {
+                        headers[binarycase(h, i + 1)] = value
```

Comment:
Well, to be honest, if someone relies on the even dirtier way of compacting headers with a comma, sure - not compatible and it would warrant a major version bump. But it brings great value to be honest.